pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -856,7 +856,7 @@ func TestGarbageCollectorSync(t *testing.T) {
 	stopCh := make(chan struct{})
 	defer close(stopCh)
 	go gc.Run(1, stopCh)
-	go gc.Sync(fakeDiscoveryClient, 10*time.Millisecond, stopCh)
```

Comment:
IIUC, the original 100x difference was to ensure the gc sync was run ~100 times to catch any flakiness.

How about setting it 200ms here, and let the test run 10s (by setting the sleep time to 10s), then the sync got tested 50 times?

Also note that the old test wasn't as reliable as it intended to be. Although the sync period was set to 10ms, because the old `WaitFor` function didn't handle the closed `done` channel, the WaitForCacheSync returned 100ms later, after the first poll period. So the old test only run the sync behavior 10 times in the 1s test time.

Reply:
> the original 100x difference was to ensure the gc sync was run ~100 times to catch any flakiness

it was actually to make sure the test waited waaaaay longer than the resync period to ensure there was time to complete at least 2 iterations

Reply:
The pseudo-code of `GarbageCollector.Sync()`:
```
// In this case, `stopCh` won't be closed. Ignore it.
GarbageCollector.Sync():
    wait.Until() loops with period:
        wait.PollImmediateUntil() loops with 100ms (hardcode):
            controller.WaitForCacheSync() loops with a channel which will be closed after the `period`
            This loop never returns unless the sync is finished.
```

1 second of period makes controller.WaitForCacheSync() tries ~10 times to check if the cache is synced.

200ms of period makes ~2 executions of `controller.WaitForCacheSync()` in every wait.PollImmediateUntil() loop. Finally `controller.WaitForCacheSync()` also executes about 10 times. (Due to the behavior of time.Ticker, it may drop some ticks for slow receivers. So the 10 is the upper bound)

Reply:
ok

Reply:
that's probably worth documenting in a comment in the test

Reply:
Done